### DISCUSSION TITLE
How can I access the URL of the project from SQL?

### DISCUSSION THREAD

**Discussion by arnu515** [supabase/supabase#15498] [Questions] [answered]
I want to add a record in a table that has a column which will point to the URL of an image in supabase storage. There is a way to get the public URL of the image `https://SUPABASE_URL/storage/v1/object/public/filepath`, but I can't hardcode the URL in SQL functions because it will not work for local development.

Is there any way to get the Supabase project URL, or atleast the project ID from SQL editor? (I'm using a procedure)

**Comment by GaryAustin1** [comment, marked as answer]
No.  You would need to store by "hand" in a table or use Vault and store it there.
Can you not just store the path and then use that in the client code to generate the image url as you have the project URL there?

**Reply by arnu515** [reply]
> Can you not just store the path and then use that in the client code to generate the image url as you have the project URL there?

I use a database function to automatically populate the table, so I can't use client code. But thanks for the first suggestion, I'll just implement that.
